### PR TITLE
chore(cts): report benchmark results

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

Report benchmark results in the generation comment directly

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -474,6 +480,25 @@ jobs:
...
 
+      - name: Read benchmark results
+        id: benchmark
+        run: |
+          # merge all benchmark results into a single json, and remove the files
+          results=$(jq -s 'add' -c tests/output/**/benchmarkResult.json)
+          {
+            echo 'BENCHMARK_SECTION<<EOF'
+            echo "<details>"
+            echo "<summary>📊 Benchmark results</summary>"
```
Comment (shortcuts) [inline .github/workflows/check.yml:491]: can we add a small disclaimer like

`take those results with a grain of salt, best effort benchmarking solution, in order to highlight general performances of our Search API`

or something like that? just so people don't expect those to be a source of truth or things like that

Comment (shortcuts) [reply .github/workflows/check.yml:491]: also we should mention that it's only tested on search?

**Review by shortcuts — COMMENTED** [review]
wonderful

File: .github/workflows/check.yml (hunk 1 above)
Comment (millotp) [reply .github/workflows/check.yml:491]: added a small disclaimer, I would like to add more benchmark in the future so I'm not very specific.

File: .github/workflows/check.yml [hunk 2]
```diff
@@ -474,6 +480,26 @@ jobs:
...
+            echo "" # empty line is required to make the table work
```
Comment (shortcuts) [inline .github/workflows/check.yml:492]: do we still need it now that we have a comment?

Comment (millotp) [reply .github/workflows/check.yml:492]: need what ?

Comment (millotp) [reply .github/workflows/check.yml:492]: Screenshot 2024-07-22 at 14 43 11

yes otherwise it looks like this

Comment (shortcuts) [reply .github/workflows/check.yml:492]: ahhh ok make sense

**Review by shortcuts — APPROVED** [review]
image

**Pull request merged**